tests/bluetooth/df/connection_cte_tx_params/src/test_set_conn_cte_tx_params.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stddef.h>
+#include <ztest.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <sys/byteorder.h>
+#include <host/hci_core.h>
+
+#include <bt_conn_common.h>
+#include "test_set_conn_cte_tx_params.h"
+
+static uint16_t g_conn_handle;
+
+static uint8_t g_ant_ids[] = { 0x1, 0x2, 0x3, 0x4, 0x5 };
+
+/**
+ * @brief Structure to store CTE receive parameters for unit tests setup.
+ */
+struct ut_bt_df_conn_cte_tx_params {
+	uint8_t cte_types;
+	uint8_t switch_pattern_len;
+	uint8_t *ant_ids;
+};
+
+static struct ut_bt_df_conn_cte_tx_params g_params;
+
+/* Macros delivering common values for unit tests */
+#define CONN_HANDLE_INVALID (CONFIG_BT_MAX_CONN + 1)
+#define ANTENNA_SWITCHING_SLOT_INVALID 0x3 /* BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US + 1 */
+#define CTE_TYPE_NONE_ALLOWED 0x0 /* Any bit is not set */
+#define CTE_TYPE_INVALID                                                                           \
+	(~(uint8_t)(BT_HCI_LE_AOA_CTE_RSP | BT_HCI_LE_AOD_CTE_RSP_1US |                            \
+		    BT_HCI_LE_AOD_CTE_RSP_2US)) /* Any other than allowed bit is set */
+
+/* Antenna switch pattern length is stored in 1 octet. If BT Core spec. extends the max value to
+ * UINT8_MAX expected failures may not be checked. If storage size is increased, tests shall be
+ * updated.
+ */
+BUILD_ASSERT(CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN < UINT8_MAX,
+	     "Can't test expected failures for switch pattern length longer or "
+	     "equal to UINT8_MAX.");
+#define SWITCH_PATTERN_LEN_TOO_LONG (CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN + 1)
+
+BUILD_ASSERT(BT_HCI_LE_SWITCH_PATTERN_LEN_MIN > 0x0,
+	     "Can't test expected failures for switch pattern length smaller or equal to zero.");
+#define SWITCH_PATTERN_LEN_TOO_SHORT (BT_HCI_LE_SWITCH_PATTERN_LEN_MIN - 1)
+
+static int send_set_conn_cte_tx_params(uint16_t conn_handle,
+				       const struct ut_bt_df_conn_cte_tx_params *params)
+{
+	struct bt_hci_cp_le_set_conn_cte_tx_params *cp;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_CONN_CTE_TX_PARAMS,
+				sizeof(*cp) + params->switch_pattern_len);
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+
+	cp->handle = sys_cpu_to_le16(conn_handle);
+	cp->cte_types = params->cte_types;
+
+	uint8_t *dest_ant_ids = net_buf_add(buf, params->switch_pattern_len);
```

Comment:
Consider moving the declaration to top of scope as has been done for other declarations in this file.

Reply:
Done, but I prefer C++ approach, declare as close as possible to place where a variable is used.

Reply:
> Done, but I prefer C++ approach, declare as close as possible to place where a variable is used.

Given that we apparently don't have any rules for that, and that there is some different opinions on it, it was fine to keep as is, but then the `err` declarations should have been moved instead :)

If there are no rules, then at least we can be consistent :D